source/blog/042-wide-gamut-colors-in-sass.md

```diff
@@ -0,0 +1,279 @@
+
+For legacy colors, the method is optional. But for non-legacy colors, a method is required. In most cases, the method can simply be a color space name. But when we're using a color space with 'polar hue' channel (such as `hsl`, `hwb`, `lch`, or `oklch`) we can also specify the *direction* we want to move around the color wheel: `shorter hue`, `longer hue`, `increasing hue`, or `decreasing hue`:
+
+```scss
+@use 'sass:color';
+
+// result: #660099
+$legacy: color.mix(red, blue, 40%);
+
+// result: ???
+$lab: color.mix(red, blue, 40%, lab);
+
+// result: ???
+$oklch-longer: color.mix(red, blue, 40%, oklch longer hue);
+```
+
+In this case, the first color in the mix is considered the 'origin' color. Like the other functions above, we can use different spaces for mixing, but the result will always be returned in that origin color space.
+
+## Working with gamut boundaries
+
+So what happens when you go outside the gamut of a given display? Browsers are still debating the details, but everyone agrees we have to display *something*:
+
+- Currently, browsers convert every color into `red`, `green`, and `blue` channels for display. If any of those channels are too high or two low for a given screen, they get *clamped* at the highest or lowest value allowed. This is often referred to as 'channel clipping'. It keeps the math simple, but it can have a weird effect on both the `hue` and `lightness` if some channels are clipped more than others.
+- The CSS specification says that preserving `lightness` should be the highest priority, and provides an algorithm for reducing `chroma` until the color is in gamut. That's great for maintaining readable text, but it's more work for browsers, and it can be surprising when colors suddenly lose their vibrance.
+- There's been some progress on a compromise approach, reducing `chroma` to get colors inside the `rec2020` gamut, and clipping from there.
+
+Since browser behavior is still unreliable, and some color spaces (*cough* `oklch`) can easily launch us out of any available gamut, it can be helpful to do some gamut management in Sass.
+
+We can use `color.is-in-gamut()` to test if a particular color is in a given gamut. Like our other color functions, this will default to the space the color is defined in, but we can provide a `$space` parameter to test it against a different gamut:
+
+```scss
+@use 'sass:color';
+$extra-pink: color(display-p3 0.951 0.457 0.7569);
+
+// result: true, for display-p3 gamut
+$in-p3: color.is-in-gamut($extra-pink);
+
+// result: false, for srgb gamut
+$in-srgb: color.is-in-gamut($extra-pink, srgb);
```

Comment:
For consistency, I think it makes sense to always recommend `$space:` be passed by name.

```suggestion
$in-srgb: color.is-in-gamut($extra-pink, $space: srgb);
```